src/main/host/host.rs

```diff
@@ -171,6 +172,15 @@ impl Host {
         crate::core::logger::log_wrapper::c_to_rust_log_level(level).map(|l| l.to_level_filter())
     }
 
+    pub unsafe fn upstream_router(&self) -> *mut Router {
```

Comment:
`unsafe` not needed? The `unsafe` operation of dereferencing the pointer is left to the caller.

Reply:
Removed.